### ISSUE TITLE
Explanation for Shap Deep Explainer Binary classification

### ISSUE THREAD

**Issue by Mahima-ai** [shap/shap#1659]
Hey @slundberg ,

I am working on a TF2.3 and using Keras binary classification model. To get the explanations for the model I am using Shap Deep Explainer.

Expected Output: The shap values are generated for both the classes, i.e. class 0 and class 1 as in the case of a multi-class classification TF Keras model using Shap Deep Explainer or a binary classification scikit model using shap Kernel Explainer.

Scenario 1:
The model is a binary classification TF model with 2 neurons and softmax activation in the output layer. 
Actual Output: As expected (The shap values are generated for both the classes, i.e. class 0 and class 1)

Scenario 2: 
The model is a binary classification TF model with 1 neuron and sigmoid activation in the output layer. 
**Actual Output: Not as expected (The shap values are generated only for one class)**

Reasoning:
The predict function of TF gives a different type of output in both scenarios.
For Scenario 1, the model.predict() gives the probability values for both classes.
For Scenario 2, the model.predict() gives a single probability value between 0 to 1, and then an extra step is required to get the probability values for both the classes. This extra step is not included in TF2.3 and we have to do it explicitly.

Question:
Is my reasoning for the shap values correct? If yes, then where does this extra step for getting the probability values for both the classes should be implemented. If no, then please explain the output of shap values for scenario 2.

Thanks.

**Comment by chengjianhong** [comment]
@slundberg I have the same question. When the features are entered into the Multilayer perception (MLP) classifier, how to get the shape values?
